src/snowflake/snowpark/modin/plugin/_internal/utils.py

```diff
@@ -1493,7 +1550,10 @@ def convert_str_to_timedelta(x: str) -> pd.Timedelta:
             # multiple timezones. So here we cast the index to the index_type when ret = pd.Index(...) above cannot
             # figure out a non-object dtype. Note that the index_type is a logical type may not be 100% accurate.
             if is_object_dtype(ret.dtype) and not is_object_dtype(index_type):
-                ret = ret.astype(index_type)
+                try:
+                    ret = ret.astype(index_type)
+                except ValueError:  # e.g., Tz-aware datetime.datetime cannot be converted to datetime64
+                    pass
```

Comment:
In this case thoughts on using `tz_localize(None)` for naive local time or `tz_convert(None)` for UTC time, then calling `astype`. Might be better than catching and passing the ValueError